File: algorithms/048_rotate_image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        if (matrix.empty()) return;
        for (int i=0;i<matrix.size()/2;i++) {
            for (int j=0;j<matrix.size();j++) {
                int temp=matrix[i][j];
                matrix[i][j]=matrix[matrix.size()-i-1][j];
                matrix[matrix.size()-i-1][j]=temp;
            }
        }
        for (int i=0;i<matrix.size();i++) {
            for (int j=i+1;j<matrix.size();j++) {
                int temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
            }
};

File: algorithms/python3/048_rotate_image.py
class Solution:
    def rotate(self, matrix):
        """
        Do not return anything, modify matrix in-place instead.
        """
        for i in range(len(matrix)):
            for j in range(0,i):
                matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]

        for i in range(len(matrix)):
            for j in range(len(matrix[0])//2):
                cur=len(matrix[0])-1-j
                matrix[i][j],matrix[i][cur]=matrix[i][cur],matrix[i][j]


a=Solution()

nums=[[1,2,3],[4,5,6],[7,8,9]]

a.rotate(nums)